### PR TITLE
Add CLUSTER SLOT-STATS document.

### PR DESCRIPTION
**Pull request by kyle-yh-kim**
Docs for CLUSTER SLOT-STATS, with key-count, cpu-usec, network-bytes-in, and network-bytes-out metrics.
- https://github.com/valkey-io/valkey/pull/351
- https://github.com/valkey-io/valkey/pull/712
- https://github.com/valkey-io/valkey/pull/720

### CODE REVIEW

File: commands/cluster-slot-stats.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
+`CLUSTER SLOT-STATS` returns an array of slot usage statistics for slots assigned to the current shard.
+The command is suitable for Valkey Cluster users aiming to assess general slot usage trends, identify hot / cold slots, migrate slots for a balanced cluster workload, and / or re-write application logic to better utilize slots.
+
+As of now, the following metrics are supported:
+* `key-count` - Number of keys residing in a given slot.
+* `cpu-usec` - Amount of cpu time (in micro-seconds) spent on a given slot.
+* `network-bytes-in` - Amount of network ingress (in bytes) spent on a given slot.
+* `network-bytes-out` - Amount of network egress (in bytes) spent on a given slot.
```
Comment (madolson) [inline commands/cluster-slot-stats.md:8]:
```suggestion
* `network-bytes-in` - Amount of network ingress (in bytes) received for given slot.
* `network-bytes-out` - Amount of network egress (in bytes) sent for a given slot.
```

File: commands/cluster-slot-stats.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
+`CLUSTER SLOT-STATS` returns an array of slot usage statistics for slots assigned to the current shard.
+The command is suitable for Valkey Cluster users aiming to assess general slot usage trends, identify hot / cold slots, migrate slots for a balanced cluster workload, and / or re-write application logic to better utilize slots.
+
+As of now, the following metrics are supported:
+* `key-count` - Number of keys residing in a given slot.
+* `cpu-usec` - Amount of cpu time (in micro-seconds) spent on a given slot.
+* `network-bytes-in` - Amount of network ingress (in bytes) received for given slot.
+* `network-bytes-out` - Amount of network egress (in bytes) sent out for given slot.
+
+## Supported arguments
```
Comment (madolson) [inline commands/cluster-slot-stats.md:10]:
```suggestion
## Supported filtering and ordering arguments
```

File: commands/cluster-slot-stats.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+      4# "network-bytes-out" => (integer) 0
+2) 1) (integer) 1
+   2) 1# "key-count" => (integer) 0
+      2# "cpu-usec" => (integer) 0
+      3# "network-bytes-in" => (integer) 0
+      4# "network-bytes-out" => (integer) 0
+```
+
+### ORDERBY
+Orders slot statistics based on the provided metric.
```
Comment (madolson) [inline commands/cluster-slot-stats.md:57]:
```suggestion
Orders slot statistics based on the specified statistic.
```

File: commands/cluster-slot-stats.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+      4) (integer) 0
+      5) "network-bytes-in"
+      6) (integer) 0
+      7) "network-bytes-out"
+      8) (integer) 0
+```
+
+##### Response in RESP3
+```
+> CLUSTER SLOT-STATS ORDERBY KEY-COUNT LIMIT 2 DESC
```
Comment (madolson) [inline commands/cluster-slot-stats.md:88]: We don't describe the limit or ASC/DESC tokens anywhere. We should probably do that.

File: commands/cluster-slot-stats.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+   2) 1# "key-count" => (integer) 0
+      2# "cpu-usec" => (integer) 0
+      3# "network-bytes-in" => (integer) 0
+      4# "network-bytes-out" => (integer) 0
+```
+
+### ORDERBY
+Orders slot statistics based on the provided metric.
+The `ORDERBY` argument allows for the user to identify hot / cold slots across the cluster.
+The response is ordered based on sort result.
```
Comment (madolson) [inline commands/cluster-slot-stats.md:59]: I think we can maybe just merge this into the first sentence, this feels a bit obvious and duplicative.

File: commands/cluster-slot-stats.md [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+The command is suitable for Valkey Cluster users aiming to assess general slot usage trends, identify hot / cold slots, migrate slots for a balanced cluster workload, and / or re-write application logic to better utilize slots.
+
+As of now, the following metrics are supported:
+* `key-count` - Number of keys residing in a given slot.
+* `cpu-usec` - Amount of cpu time (in micro-seconds) spent on a given slot.
+* `network-bytes-in` - Amount of network ingress (in bytes) received for given slot.
+* `network-bytes-out` - Amount of network egress (in bytes) sent out for given slot.
+
+## Supported arguments
+There exist two mutually exclusive arguments, namely;
```
Comment (madolson) [inline commands/cluster-slot-stats.md:11]:
```suggestion
There exist two mutually exclusive arguments for controlling the output, namely;
```

File: commands/cluster-slot-stats.md [hunk 7]
```diff
@@ -0,0 +1,99 @@
+`CLUSTER SLOT-STATS` returns an array of slot usage statistics for slots assigned to the current shard.
+The command is suitable for Valkey Cluster users aiming to assess general slot usage trends, identify hot / cold slots, migrate slots for a balanced cluster workload, and / or re-write application logic to better utilize slots.
+
+As of now, the following metrics are supported:
```
Comment (madolson) [inline commands/cluster-slot-stats.md:4]:
```suggestion
The following metrics are supported:
```

File: commands/cluster-slot-stats.md [hunk 8]
```diff
@@ -0,0 +1,99 @@
+`CLUSTER SLOT-STATS` returns an array of slot usage statistics for slots assigned to the current shard.
```
Comment (madolson) [inline commands/cluster-slot-stats.md:1]: I just realize we kind of use metrics and statistics sort of interchangeably which now bothers me, but maybe it's fine.

**Review by madolson — COMMENTED** [review]
Sorry for the delay, finally really looked through everything and left a few comments about clarity.

**Comment by zuiderkwast** [comment]
@kyle-yh-kim Will you update this PR? It would be good to have docs for the feature. 8.0 is already released.

File: commands.json [hunk 9]
```diff
@@ -3095,6 +3095,87 @@
             "nondeterministic_output"
         ]
     },
+    "CLUSTER SLOT-STATS": {
```
[the remark this answers is not in the document]
Comment (zuiderkwast) [reply commands.json]: From the JSON file in the code repo.

**Review by zuiderkwast — COMMENTED** [review]
Thanks! I have some comments about the structure of the page.

When we generate the man pages, stuff like Reply, ACL categories and History are inserted above Examples. (It looks for certain headings.) See example:

image

File: commands/cluster-slot-stats.md (hunk 8 above)
Comment (zuiderkwast) [reply commands/cluster-slot-stats.md:1]: The command name is the title of the page, so it doesn't need to here.

```suggestion
Returns an array of slot usage statistics for slots assigned to the current shard.
```

File: commands/cluster-slot-stats.md [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+* `network-bytes-in` - Amount of network ingress (in bytes) received for given slot.
+* `network-bytes-out` - Amount of network egress (in bytes) sent out for given slot.
+
+## Supported filtering and ordering arguments
+There exist two mutually exclusive arguments for controlling the output, namely;
+
+### SLOTSRANGE
+Returns slot statistics based on the slots range provided. The range is inclusive.
+The `SLOTSRANGE` argument allows for request pagination.
+The response is ordered in ascending slot number.
```
Comment (zuiderkwast) [inline commands/cluster-slot-stats.md:16]: The syntax is also inserted automatically above this text, from the JSON file. We should document the options. If they're mutually exclusive, the syntax will display this as `< SLOTRANGE xxxx | ORDERBY yyy >` so I don't think we need to mention that.

```suggestion
## Options

* `SLOTSRANGE`: Returns slot statistics based on the slots range provided.
  The range is inclusive.
  The `SLOTSRANGE` argument allows for request pagination.
  The response is ordered in ascending slot number.
```

File: commands/cluster-slot-stats.md [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+      2# "cpu-usec" => (integer) 0
+      3# "network-bytes-in" => (integer) 0
+      4# "network-bytes-out" => (integer) 0
+2) 1) (integer) 13902
+   2) 1# "key-count" => (integer) 20
+      2# "cpu-usec" => (integer) 0
+      3# "network-bytes-in" => (integer) 0
+      4# "network-bytes-out" => (integer) 0
+```
+```
```
Comment (madolson) [inline commands/cluster-slot-stats.md:78]:
```suggestion
```

File: commands/cluster-slot-stats.md [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+   2) 1) "key-count"
+      2) (integer) 20
+      3) "cpu-usec"
+      4) (integer) 0
+      5) "network-bytes-in"
+      6) (integer) 0
+      7) "network-bytes-out"
+      8) (integer) 0
+```
+```
```
Comment (madolson) [inline commands/cluster-slot-stats.md:41]:
```suggestion
```
Examples typically flow.

File: commands/cluster-slot-stats.md [hunk 13]
```diff
@@ -0,0 +1,90 @@
...
+The following statistics are supported:
+
+* `key-count` -- Number of keys residing in a given slot.
+* `cpu-usec` -- Amount of CPU time (in microseconds) spent on a given slot.
+* `network-bytes-in` -- Amount of network ingress (in bytes) received for given slot.
+* `network-bytes-out` -- Amount of network egress (in bytes) sent out for given slot.
+
+## Options
+
+* `ORDERBY` -- Returns an ordered slot statistics based on the specified statistic and sub-arguments to identify hot / cold slots across the cluster. In the event of a tie in the stats, ascending slot number is used as a tie breaker.
```
Comment (madolson) [inline commands/cluster-slot-stats.md:13]: In the past we talked about also including the descriptions for limit and ASC/DESC. Specifically to implement pagination if someone wanted to do that.

File: commands/cluster-slot-stats.md (hunk 12 above)
Comment (zuiderkwast) [reply commands/cluster-slot-stats.md:41]: They are two different examples?

If yes, then I would add some text between them to say what's what.

**Comment by zuiderkwast** [comment]
Good, but add it to resp3 replies too, please. Usually all commands are in both files. (If the content is identical, the man page will detect that and not include it twice.)

**Review by zuiderkwast — APPROVED** [review]
Great! I'm happy now.

**Comment by zuiderkwast** [comment]
And Thank you for completing this!

**Pull request merged**